### PR TITLE
chore: Refactor create function app

### PR DESCRIPTION
**Pull request by AhmedAbdoOrtiga**
#### Description of changes
- Merge deploy arm template code for web worker and wep api in one function.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/resource-deployment/scripts/function-app-create.sh [hunk 1]
```diff
@@ -123,58 +123,30 @@ getFunctionAppPrincipalId() {
...
+    myWebFunctionAppName="$resourceName"
 
-    waitForFunctionAppServiceDeploymentCompletion $webApiFunctionAppName
-    echo "Successfully deployed Azure Function App '$webApiFunctionAppName'"
-}
-
-deployWebWorkersArmTemplate() {
-    functionAppNamePrefix="web-workers-allyfuncapp"
-    templateFilePath="${0%/*}/../templates/function-web-workers-app-template.json"
+    waitForFunctionAppServiceDeploymentCompletion $myWebFunctionAppName
```
Comment (devaradhanm) [inline packages/resource-deployment/scripts/function-app-create.sh]: just name it functionAppName

File: packages/resource-deployment/scripts/function-app-create.sh [hunk 2]
```diff
@@ -109,72 +109,44 @@ publishFunctionAppScripts() {
...
     . "${0%/*}/key-vault-enable-msi.sh"
-}
 
-deployWebWorkersFunctionApp() {
-    deployWebWorkersArmTemplate $webWorkersPackageName
-
-    # Keep child script call only one level deep to preserve exports
-    getFunctionAppPrincipalId $webWorkersFunctionAppName
-    . "${0%/*}/key-vault-enable-msi.sh"
+    eval $functionAppName="'$myFunctionAppName'"
```
Comment (devaradhanm) [inline packages/resource-deployment/scripts/function-app-create.sh]: why not just functionAppName="'$myFunctionAppName'" ?

Comment (devaradhanm) [reply packages/resource-deployment/scripts/function-app-create.sh]: also why is this line needed? who is using it?

Comment (AhmedAbdoOrtiga) [reply packages/resource-deployment/scripts/function-app-create.sh]: I don't want to set functionAppName, I want to set it's value, this is one way to handle Returning Values from Bash Functions
so the command (eval $functionAppName="'$myFunctionAppName'") will be compiled as 
(eval webAPiFunctionAppName="'$myFunctionAppName'") for webApi call and as
(eval webWorkerFunctionAppName="'$myFunctionAppName'") for webWorker call

Look at this: https://www.linuxjournal.com/content/return-values-bash-functions

Comment (devaradhanm) [reply packages/resource-deployment/scripts/function-app-create.sh]: P2: i see. you could use get-resource-names.sh to get the names.

**Pull request merged**